e2e/tests/core/02-client/client_test.go

```diff
@@ -267,7 +267,12 @@ func (s *ClientTestSuite) TestAllowedClientsParam() {
 
 	t.Run("ensure allowed clients are set to the default", func(t *testing.T) {
 		allowedClients := s.QueryAllowedClients(ctx, chainA)
-		s.Require().Equal(clienttypes.DefaultAllowedClients, allowedClients)
+
+		defaultAllowedClients := clienttypes.DefaultAllowedClients
+		if !testvalues.LocalhostClientFeatureReleases.IsSupported(chainAVersion) {
+			defaultAllowedClients = defaultAllowedClients[:len(defaultAllowedClients)-1]
```

Comment:
this only works since currently the localhost client is the last entry in this list. I think we should explicitly remove the one we care about. Maybe something like

```golang
func Remove[T comparable](elem T, elements []T) []T {
	var elementsCopy []T
	for i, e := range elements {
		if elem != e {
			elementsCopy = append(elementsCopy, elements[i])
		}
	}
	return elementsCopy
}


defaultAllowedClients = Remove(ibcexported.Localhost, allowedClients)
```

To explicitly remove the localhost.

Reply:
since we're on go 1.21, there's [`slices.DeleteFunc`](https://pkg.go.dev/slices@master#DeleteFunc) that we could also use

Reply:
even better!